app/models/user.rb

```diff
@@ -1,4 +1,13 @@
 class User < ApplicationRecord
   validates_presence_of :name, :email, :token, :google_id
   validates :email, uniqueness: true
+
+  def self.find_or_create_by_auth(auth_data)
+    user = self.find_or_create_by_provider_and_uid(auth_data["provider"], auth_data["uid"])
```

Comment:
Very curious to know what this code is doing, it looks pretty cool! I don't see it being called anywhere in our code. Is this something that will be helpful later on?